mllib/src/main/scala/org/apache/spark/ml/param/shared/SharedParamsCodeGen.scala

```diff
@@ -97,6 +97,7 @@ private[shared] object SharedParamsCodeGen {
       doc: String,
       defaultValueStr: Option[String] = None,
       isValid: String = "",
+      finalFields: Boolean = true,
```

Comment:
Can we combine ```finalFields``` and ```finalMethods``` as ```is FinalParam```? If the set/get methods are non-final, it makes sense to get a non-final param.